### PR TITLE
Byte array was fixed; Unused import is removed

### PR DESCRIPTION
**Pull request by solomax**
Input byte sequence was broken (c3 byte was inserted into strings 2 times)
I have corrected this, but unfortunately have no idea why this extra byte was added :(

### CODE REVIEW

**Comment by mondain** [comment]
I believe the "chunk size" is supposed to start at 128 and then be increased or decreased based on chunk size messages.

**Comment by solomax** [comment]
Wasn't aware of this :(
Then this fix need to be improved

**Comment by mondain** [comment]
I was attempting to fix or optimize the in buffer handling; I could see 00 00 09 in the first set of bytes that came in with or including the connect data and i wondered why in the heck they were only reading chunks of the data at a time, when it was already all there and maybe more. BufferUtils definitely doesn't seem like its efficient at all for its put method imho.

**Comment by mondain** [comment]
When my decoding fails after I play a second video in oflaDemo, there seems to be 12 or so bytes "stuck" in the buffers and I think that is the root of that issue.

**Comment by mondain** [comment]
From what I understand the "Header" data is often or possibly always 12 bytes long.

**Comment by solomax** [comment]
Is it still reproducible with this fix?

**Comment by mondain** [comment]
Didn't try it yet, but I think locking down to 128 will fail when the chunk size changes; I'm not sure which end controls the changes, It may be both endpoints.

**Comment by solomax** [comment]
Could you please try it?
Fixed chunk size can be fixed later

File: src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java [hunk 1]
```diff
@@ -325,6 +335,8 @@ public Packet decodePacket(RTMPConnection conn, RTMPDecodeState state, IoBuffer
         buf.flip();
         try {
             final IRTMPEvent message = decodeMessage(conn, packet.getHeader(), buf);
+            //we have just decoded full buf, need to advance original IoBuffer
+            in.skip(correction + buf.position());
```
Comment (solomax) [inline src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java:345]: I believe this was one of the main issues with packet decoding
original buffer "in" was not advanced after decoding of "buf" buffer, so it was being "re-parsed" with part of header cut-off

File: src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java [hunk 2]
```diff
@@ -313,9 +316,16 @@ public Packet decodePacket(RTMPConnection conn, RTMPDecodeState state, IoBuffer
...
-        //BufferUtils.put(buf, in, readAmount);
-        if (buf.position() < header.getSize()) {
+        int correction = 0;
+        //we will fill the buffer chunk by chunk skipping any CHUNK_DELIMITER found
+        for (int i = in.position(); i < in.position() + readAmount; i += CHUNK_SIZE) {
+        	if (in.array()[i] == CHUNK_DELIMITER) {
+        		i++;
+        		correction++;
+        	}
+    		buf.put(in.array(), i, Math.min(CHUNK_SIZE, in.position() + readAmount + correction - i));
```
Comment (solomax) [inline src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java]: This part of code helps to remove 'C3' chunk delimiter bytes
the constant size here can easily be replaced with variable size

File: src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java (hunk 1 above)
Comment (mondain) [inline src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java:345]: Theres a conflict with head, so I'll apply this by-hand and test it out.

**Comment by mondain** [comment]
Ok, I added and tested, it fails on the 3rd packet decode now. I updated the test with the latest dumps.

**Comment by solomax** [comment]
have manually merged it back
will try to check tests today

**Comment by solomax** [comment]
It seems like header for 3rd packet is broken
Am I right thinking decodeBuffer receives full buffer, then passes part of it to decodePacket?
Maybe we can get full buffer to find the place buffer is being corrupted?

**Comment by mondain** [comment]
I'll have to confirm, but I think decodeBuffer was used in the old
handshake routine; it may no longer be used.

**Pull request closed without merging**